WordPress/Classes/ViewRelated/Blog/LanguageViewController.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+import WordPressShared
+
+
+/// This class will display the Blog's Language setting, and will allow the user to pick a new value.
+/// Upon selection, WordPress.com backend will get hit, and the new value will be persisted.
+///
+public class LanguageViewController : UITableViewController
+{
+    /// Callback to be executed whenever the Blog's selected language changes.
+    ///
+    var onChange : (NSNumber -> Void)?
+    
+    
+    
+    /// Designated Initializer
+    ///
+    /// - Parameters
+    ///     - Blog: The blog for which we wanna display the languages picker
+    ///
+    public convenience init(blog: Blog) {
+        self.init(style: .Grouped)
+        self.blog = blog
+    }
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+    
+        // Setup tableViewController
+        title = NSLocalizedString("Language", comment: "Title for the Language Picker Screen")
+        clearsSelectionOnViewWillAppear = false
+    
+        // Setup tableView
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+        WPStyleGuide.resetReadableMarginsForTableView(tableView)
+    }
+    
+    public override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Reload + Clear selection. Let's avoid flickers by dispatching the deselect call asynchronously
+        tableView.reloadDataPreservingSelection()
+        
+        dispatch_async(dispatch_get_main_queue()) {
+            self.tableView.deselectSelectedRowWithAnimation(true)
+        }
+    }
+    
+    
+    
+    // MARK: - UITableViewDataSource Methods
+    public override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    public override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    
+    public override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier)
```

Comment:
Reason for not doing that is: because we need to initialize the `WPTableViewCell` instance with a custom style, which is not the default. 

The register + dequeue mechanism doesn't allow us to specify a custom style, and i'd really love to avoid adding yet another subclass.